### ISSUE TITLE
`sqleton serve`: Watcher stops watching after moving a file

### ISSUE THREAD

**Issue by sebcode** [go-go-golems/sqleton#154]
Repro:

Run in terminal 1:
```
cd /tmp
mkdir -p /tmp/sql/test
echo 'repositories:' > /tmp/repo.yml
echo '- /tmp/sql' >> /tmp/repo.yml
sqleton --config /tmp/repo.yml serve --content-dirs /tmp/sql/ --log-level=DEBUG
```

Run in terminal 2:
```
cd /tmp/sql/test; for i in 1 2 3; do echo 1>a.yaml.tmp; mv a.yaml.tmp a.yaml; done
```

After the 3rd rename the watcher seems to remove all watch paths. After that, no files are being watched anymore:
```
10:26AM DBG Received fsnotify event event="CHMOD         \"\""
10:26AM DBG Skipping event because it does not match the mask masks=["**/*.yaml"] path=
10:26AM DBG Received fsnotify event event="REMOVE        \"\""
10:26AM DBG Removing paths with prefix name= watchlist=["/tmp/sql/test/a.yaml","/tmp/sql","/tmp/sql/test"]
```

Happens with v0.1.53

```
$ uname -a
Linux rz 6.2.0-20-generic #20-Ubuntu SMP PREEMPT_DYNAMIC Thu Apr  6 07:48:48 UTC 2023 x86_64 x86_64 x86_64 GNU/Linux
```

**Comment by wesen** [comment]
Addressed (kind of) in https://github.com/go-go-golems/clay/pull/16

**Comment by wesen** [comment]
Fixed in v0.1.55

**Comment by wesen** [comment]
v0.1.56 actually
